x-pack/plugin/sql/qa/src/main/resources/datetime.csv-spec

```diff
@@ -612,6 +665,15 @@ null    | 1999-04-30 00:00:00.000Z
 F       | 1997-05-19 00:00:00.000Z
 ;
 
+dateTruncHavingWithInterval
+schema::gender:s|dt:ts
+SELECT gender, max(hire_date) AS dt FROM test_emp GROUP BY gender HAVING max(hire_date) - DATE_TRUNC('hour', INTERVAL 1 YEARS) >= '1997-01-01T00:00:00.000Z'::timestamp ORDER BY 1;
```

Comment:
@astefan @matriv 
I am working on this query. 
> HAVING max(hire_date) - [ANY INTERVAL VALUE like INTERVAL 1 YEARS]
I am not using DATE_TRUNC function in the test and I get this error.

`Caused by: java.lang.ClassCastException: class org.elasticsearch.xpack.sql.expression.literal.interval.IntervalYearMonth cannot be cast to class java.lang.Number (org.elasticsearch.xpack.sql.expression.literal.interval.IntervalYearMonth is in unnamed module of loader java.net.FactoryURLClassLoader @6a3fbcb1; java.lang.Number is in module java.base of loader 'bootstrap')	at org.elasticsearch.xpack.sql.expression.predicate.operator.arithmetic.SqlBinaryArithmeticOperation.lambda$static$1(SqlBinaryArithmeticOperation.java:60)	at org.elasticsearch.xpack.sql.expression.predicate.operator.arithmetic.SqlBinaryArithmeticOperation.doApply(SqlBinaryArithmeticOperation.java:129)	at org.elasticsearch.xpack.ql.expression.predicate.PredicateBiFunction.apply(PredicateBiFunction.java:24)	at org.elasticsearch.xpack.sql.expression.function.scalar.whitelist.InternalSqlScriptUtils.sub(InternalSqlScriptUtils.java:142)	at org.elasticsearch.painless.PainlessScript$Script.execute(InternalSqlScriptUtils.nullSafeFilter(InternalSqlScriptUtils.gte(InternalSqlScriptUtils.sub(params.a0,InternalSqlScriptUtils.intervalYearMonth(params.v0,params.v1)),InternalSqlScriptUtils.asDateTime(params.v2))):125)	... 24 more`

In my opinion, It is not related to DATE_TRUNC function. I already tried with
`SELECT gender, max(hire_date) AS dt, COUNT(*) FROM test_emp GROUP BY gender HAVING max(hire_date) - INTERVAL 1 YEARS >= '1967-01-01T00:00:00.000Z'::timestamp ORDER BY 2`
It gives same error.

Reply:
@matriv Most probably, there is a problem with max() function, because it converts max(hire_date) to Number so when I debugged, I saw that It executes 
```
if (l instanceof Number) {
 return Arithmetics.sub((Number) l, (Number) r);
}
```
in SqlBinaryArithmeticOperation class at SUB enum.
That's why it trying to cast Interval to Number.

Reply:
@musteaf Hm, yeah, doesn't seem right, maybe something wrong with the data type of max(). Would you mind opening an issue for that?

Reply:
@matriv Okay, I will open an issue. Also, I think I need to -ignore this test case.

Reply:
Yes please, and add a comment pointing to the issue, Thx!

Reply:
Hi @matriv, I added. https://github.com/elastic/elasticsearch/issues/53565. Also I want to work on it if it is possible.